### PR TITLE
Getting Started with the C/C++ Chainguard Image 

### PR DESCRIPTION
**Pull request by mcaveety**
## Type of change
Adding a new Getting Started guide to the Academy for C/C++ images!! 

### What should this PR do?
Resolves [#3888](https://github.com/chainguard-dev/internal/issues/3888)

### Why are we making this change?
Recently we published a [cheatsheet on the different uses of images for various compiled programs](https://edu.chainguard.dev/chainguard/chainguard-images/working-with-images/compiled-programs/) so this document provides an example of how we can get started using them! 

### What are the acceptance criteria? 
The guide should be functional and it should follow a similar format to the other guides we currently have. It should walk through using Chainguard Images to form a single and multi-layer docker build

### How should this PR be tested?
Please go through the guide as a learner and test the commands. Please also look for places where improvements can be made to the clarity of the writing.

Link to the guide in the deploy preview: https://deploy-preview-1723--ornate-narwhal-088216.netlify.app/chainguard/chainguard-images/getting-started/c/

### CODE REVIEW

**Comment by mcaveety** [comment]
@SharpRake This will be ready for a review whenever you get the chance. I'm also opening another PR on the edu-images-demos repository to go alongside this. I will tag you there too :)

I built and tested the site locally, as well as tested the commands, so everything seems to work well from my end.

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:270]: again, could you add a transition/closing sentence here?

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:273]:
```suggestion
With a working C++ example, you can compile and run our program using a multi-stage build. With the text editor of your choice, create a new file named `Dockerfile3`.
```

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:280]: I wanted to say: nice attention to detail with these lists! They're all formatted beautifully and identically :ok_hand:

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:311]:
```suggestion
With your new Dockerfile created, you can build the image. Execute the following command in your terminal to build your multi-stage C++ image.
```

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:317]:
```suggestion
With your image built, you can now run it with the following command.
```

I feel like I might have missed some paragraphs that split between "you" and "we" - maybe do a scan for this?

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:329]: again, add a transition sentence after this output

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (SharpRake) [inline content/chainguard/chainguard-images/getting-started/c/index.md:323]:
```suggestion
You will see output in your terminal identical to that of the C++ binary you compiled locally.
```

**Review by SharpRake — COMMENTED** [review]
This doc works great! Nice job building out these examples, I think it's all super clear and helpful.

I'm holding off on approving because, while most of my comments are very nitpicky, there are some structural things I'd like to see changed before I approve. I'm mostly thinking about closing/transition sentences between steps and sections.

File: content/chainguard/chainguard-images/getting-started/c/index.md
Comment (mcaveety) [reply content/chainguard/chainguard-images/getting-started/c/index.md:130]: I might be missing something... when I run any docker-related commands, they fail unless I manually start up Docker by opening Docker Desktop. I'll have to work out why that is 😓

**Comment by mcaveety** [comment]
I think these changes should look good 🥹 I'm so sorry for making you review all these obvious edits Mark

**Comment by mcaveety** [comment]
Going to need an extra minute here... some changes are being made to the dockerfiles

**Comment by mcaveety** [comment]
Update 8/12: Some changes are being made to the named users of the `gcc-glibc` image. Because of this, changes to the written guide need to be made. This will be reopened when these changes are ready to be reviewed.

**Comment by mcaveety** [comment]
@SharpRake Should be ready now! I hope! Thank you for being ever so patient!

**Review by SharpRake — APPROVED** [review]
This looks good to me, thanks Michelle!

**Pull request merged**
